creator-node/src/services/stateMachineManager/stateMonitoring/findReplicaSetUpdates.jobProcessor.ts

```diff
@@ -278,35 +280,32 @@ const _findReplicaSetUpdatesForUser = async (
         )
         unhealthyReplicas.add(replica.endpoint)
       } else if (unhealthyPeersSet.has(replica.endpoint)) {
-        // Else, continue with conducting extra health check if the current observed node is a primary, and
-        // add to `unhealthyReplicas` if observed node is a secondary
-        let addToUnhealthyReplicas = true
-
-        if (replica.endpoint === primary) {
-          addToUnhealthyReplicas = !(await CNodeHealthManager.isPrimaryHealthy(
-            primary
-          ))
-        }
-
-        if (addToUnhealthyReplicas) {
-          logger.error(
-            `_findReplicaSetUpdatesForUser(): Replica ${replica.endpoint} for user ${wallet} was already marked unhealthy and failed an additional health check if it was primary.`
-          )
-          unhealthyReplicas.add(replica.endpoint)
-        }
+        logger.error(
+          `_findReplicaSetUpdatesForUser(): Replica ${replica.endpoint} for user ${wallet} was already marked unhealthy.`
+        )
+        unhealthyReplicas.add(replica.endpoint)
       }
     }
   }
 
   // If any unhealthy replicas found for user, enqueue an updateReplicaSetOp for later processing
-  if (unhealthyReplicas.size > 0) {
+  const nodesToReconfigOffOf = new Set<string>()
+  for (const node of unhealthyReplicas) {
+    const shouldReconfigOffNode =
+      !(await CNodeHealthManager.isNodeHealthyOrInGracePeriod(
```

Comment:
what are your thoughts on putting this inside the isNodeHealthy function? that way it propagates downstream to unhealthyPeers (called by monitorState). consolidating all the health checks into a single place seems easier to manage vs having multiple functions which can be composed but also more confusing and possibly more error prone

Reply:
I'm leaning toward just merging as is since this works and then rethinking a bigger refactor with the queue so I have a more stable last-known-working point to revert to - wdyt?

but for this point specifically are you saying like in isNodeHealthy make it check for the grace period so unhealthyReplicas here only ever has nodes that we truly want to reconfig off of?

Reply:
for this point yes isNodeHealthy should consider the grace period and return like a bool for healthy and there should be no additional logic downstream

Reply:
whether you want to bundle that with v2 or do now up to you